optimum/onnxruntime/quantization.py

```diff
@@ -0,0 +1,262 @@
+        **kwargs
+    ):
+        """
+        Args:
+            model_name_or_path (:obj:`str`):
+                Repository name in the Hugging Face Hub or path to a local directory hosting the model.
+            output_dir (:obj:`Union[str, Path]`):
+                The output directory where the quantized model will be saved.
+            ort_config (:obj:`Union[ORTConfig, str]`):
+                Configuration file containing all the information related to the model quantization.
+                Can be either:
+                    - an instance of the class :class:`ORTConfig`,
+                    - a string valid as input to :func:`ORTConfig.from_pretrained`.
+            feature (:obj:`str`):
+                Feature used when exporting the model.
+            calib_dataset (:obj:`Dataset`, `optional`):
+                Dataset to use for the calibration step.
+            dataset_name (:obj:`str`, `optional`):
+                Dataset repository name on the Hugging Face Hub or path to a local directory containing data files to
+                load to use for the calibration step.
+            dataset_config_name (:obj:`str`, `optional`):
+                Name of the dataset configuration.
+            data_files (:obj:`str`, `optional`):
+                Path to source data files.
+            preprocess_function (:obj:`Callable`, `optional`):
+                Processing function to apply to each example after loading dataset.
+            cache_dir (:obj:`str`, `optional`):
+                Path to a directory in which a downloaded configuration should be cached if the standard cache should
+                not be used.
+            force_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to force to (re-)download the configuration files and override the cached versions if
+                they exist.
+            resume_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to delete incompletely received file. Attempts to resume the download if such a file
+                exists.
+            revision(:obj:`str`, `optional`):
+                The specific version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so ``revision`` can be any
+                identifier allowed by git.
+        """
+        config_kwargs_default = [
+            ("cache_dir", None),
+            ("force_download", False),
+            ("resume_download", False),
+            ("revision", None),
+        ]
+        config_kwargs = {name: kwargs.get(name, default_value) for (name, default_value) in config_kwargs_default}
+        self.cache_dir = config_kwargs.get("cache_dir")
+        self.model_name_or_path = model_name_or_path
+        self.output_dir = output_dir if isinstance(output_dir, Path) else Path(output_dir)
```

Comment:
Open question: Do we want to put the `output_dir` as part of the instance attributes?

We can expose this parameter on methods, to reuse the same quantizer, without being dependent on the model.

For instance: Let's imagine I have a quantization config, I would like to recover the quantizer, potentially with a different output path for the generated model.

Reply:
It's actually a good point as I feel it may be a more intuitive usage